### ISSUE TITLE
Session Manager output to S3 does not include ACL in initial upload

### ISSUE THREAD

**Issue by mithun** [aws/amazon-ssm-agent#197]
**Scenario**

1. I have an S3 bucket `mylogsbucket` in a _logs_ AWS account (_AccountA_) with the following bucket policy

```json
{
  "Sid": "",
  "Effect": "Allow",
  "Principal": {
    "AWS": [
      "arn:aws:iam::AccountB:root"
    ]
  },
  "Action": [
    "s3:PutObjectAcl",
    "s3:PutObject"
  ],
  "Resource": "arn:aws:s3:::mylogsbucket/*",
  "Condition": {
    "StringEquals": {
      "s3:x-amz-acl": "bucket-owner-full-control"
    }
  }
}
```

2. I am using SSM Session Manager in multiple AWS accounts (e.g., _AccountB_)
3. I want to save session logs from _AccountB_ in the s3 bucket in _AccountA_.
4. The `SSM-SessionManagerRunShell` in _AccountB_ is configured to save session data to `mylogsbucket`
5. The instance profile where the session is running has the following policy statement:

```json
{
  "Sid": "",
  "Effect": "Allow",
  "Action": [
    "s3:PutObjectAcl",
    "s3:PutObject"
  ],
  "Resource": [
    "arn:aws:s3:::mylogsbucket/*"
  ]
}
```

**Expected Behavior**

When a session is terminated, logs are uploaded to `mylogsbucket` with `bucket-owner-full-control` ACL.

**Current Behavior**

Logs upload to `mylogsbucket` fails with `AccessDenied`. This seems to be due to the canned ACL not provided during the initial [upload](https://github.com/aws/amazon-ssm-agent/blob/master/agent/s3util/s3util.go#L93). The current approach is to make two separate API calls to `PutObject` and `PutObjectAcl` instead of providing the canned ACL as part of the `PutObject` call.

**Related Issues**

* #139 
* #186

**Comment by sruthi-maddineni** [comment]
This is fixed in Agent version 2.3.722.0. Please update the agent.

**Comment by vector-sec** [comment]
I'm noticing this behavior with version 2.3.978.0 of the agent. I actually don't receive any logs in the S3 bucket if I have `"s3:x-amz-acl": "bucket-owner-full-control"` set in my bucket policy.
